redfish-core/lib/pcie.hpp

```diff
@@ -90,10 +101,52 @@ inline void requestRoutesSystemPCIeDeviceCollection(App& app)
                     {"Description", "Collection of PCIe Devices"},
                     {"Members", nlohmann::json::array()},
                     {"Members@odata.count", 0}};
+
                 getPCIeDeviceList(asyncResp, "Members");
             });
 }
 
+inline void getGeneration(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                          const std::string& devicePath)
+{
+    const std::string slotPath =
+        sdbusplus::message::object_path(devicePath).parent_path();
+    if (slotPath.empty())
+    {
+        BMCWEB_LOG_DEBUG << "Slot path not found";
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, slotPath](const boost::system::error_code ec,
+                              const std::variant<std::string>& property) {
+            if (ec)
+            {
+                BMCWEB_LOG_DEBUG << "DBUS response error";
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            const std::string* value = std::get_if<std::string>(&property);
+            if (value == nullptr)
+            {
+                // illegal value
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            std::string generation = analysisGeneration(*value);
+            if (!generation.empty())
+            {
+                asyncResp->res.jsonValue["PCIeInterface"]["PCIeType"] =
+                    generation;
+            }
+        },
+        std::string(pcieService), slotPath, "org.freedesktop.DBus.Properties",
+        "Get", "xyz.openbmc_project.Inventory.Item.PCIeSlot", "Generation");
```

Comment:
This should be on the Device interface, not the slot.  The DBus property is GenerationInUse.